library/types/test/hostname_test.rb

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env rspec
+
+require_relative "test_helper"
+
+require "yast"
+
+describe "Hostname#CurrentFQ" do
+  Yast.import "Hostname"
+  Yast.import "FileUtils"
+
+  let(:test_hostname) { "host.suse.cz" }
+
+  it "returns output of hostname --fqdn if available" do
+    allow(Yast::SCR)
+      .to receive(:Execute)
+      .with(path(".target.bash_output"), "hostname --fqdn")
+      .and_return({"stdout" => test_hostname, "exit" => 0 })
+
+    expect(Yast::Hostname.CurrentFQ).to eq test_hostname
+  end
+
+  it "reads /etc/hostname when hostname --fqdn fails" do
+    allow(Yast::SCR)
+      .to receive(:Execute)
+      .with(path(".target.bash_output"), "hostname --fqdn")
+      .and_return(nil)
+    allow(Yast::SCR)
+      .to receive(:Read)
+      .with(path(".target.string"), "/etc/hostname")
+      .and_return(test_hostname)
+    allow(Yast::FileUtils)
+      .to receive(:Exists)
+      .with("/etc/hostname")
+      .and_return(true)
+
+    expect(Yast::Hostname.CurrentFQ).to eq test_hostname
```

Comment:
Better, thanks.

Although being NP, I would prefer to have two different names for each case, to make more clear they are coming from different places. Something like thi
```ruby
let(:command_hostname) { "host.suse.cz" }
let(:etc_hostname) { "host.suse.de" }
```
And then
```ruby
expect(...).to eq command_hostname
...
expect(...).to eq etc_hostname
```

As said. Just NP. I can leave with the current implementation.